### PR TITLE
Catch unhandledrejection and throw error

### PR DESCRIPTION
**Pull request by smyzin**
- created global hook to catch `unhandledrejection`
- added hook to `Form`

https://user-images.githubusercontent.com/5114069/185982601-ee485afd-d0e8-4644-a7c0-1aab8c879c67.mov

### CODE REVIEW

**Review by T-Damer — APPROVED** [review]

File: src/helpers/handleError.ts [hunk 1]
```diff
@@ -24,6 +24,13 @@ export function parseError(error: unknown, defaultMessage = ErrorList.unknown) {
     const revertMessage = parseRevertReason(message)
     displayedError = gSNMessage || revertMessage || message
   }
+  if (
+    displayedError?.includes(
+      'can\'t access property "error", res2 is undefined'
+    )
+  )
+    displayedError =
+      'Relay server is not available due to a large fee cost of transaction (insufficient funds for gas * price + value). Try again later.'
```
Comment (T-Damer) [inline src/helpers/handleError.ts:33]: `Please, try again later` sounds more gentle for me :)

File: src/hooks/catchUnhandledRejection.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { useEffect } from 'preact/hooks'
+
+export default function catchUnhandledRejection(
+  callback?: (e: unknown) => void
+) {
+  useEffect(() => {
+    const listener = (event: PromiseRejectionEvent) => {
```
Comment (T-Damer) [inline src/hooks/catchUnhandledRejection.ts:7]: Destructure reason here

**Review by T-Damer — APPROVED** [review]
Lets go

File: src/hooks/catchUnhandledRejection.ts (hunk 2 above)
Comment (smyzin) [reply src/hooks/catchUnhandledRejection.ts:7]: Did it

File: src/helpers/handleError.ts (hunk 1 above)
Comment (smyzin) [reply src/helpers/handleError.ts]: Did it

File: src/helpers/handleError.ts [hunk 3]
```diff
@@ -24,6 +24,13 @@ export function parseError(error: unknown, defaultMessage = ErrorList.unknown) {
     const revertMessage = parseRevertReason(message)
     displayedError = gSNMessage || revertMessage || message
   }
+  if (
+    displayedError?.includes(
+      'can\'t access property "error", res2 is undefined'
```
Comment (T-Damer) [inline src/helpers/handleError.ts:29]: Why don't you use double quotes here? Like `"can't access..."`

**Review by T-Damer — APPROVED** [review]

**Review by MixailE — APPROVED** [review]

File: src/helpers/handleError.ts (hunk 3 above)
Comment (MixailE) [reply src/helpers/handleError.ts]: Here it seems better to take `can't access...`

**Pull request merged**